cpp/src/neighbors/detail/nn_descent.cuh

```diff
@@ -588,40 +593,46 @@ __launch_bounds__(BLOCK_SIZE)
   wmma::fragment<wmma::matrix_a, WMMA_M, WMMA_N, WMMA_K, half, wmma::row_major> a_frag;
   wmma::fragment<wmma::matrix_b, WMMA_M, WMMA_N, WMMA_K, half, wmma::col_major> b_frag;
   wmma::fragment<wmma::accumulator, WMMA_M, WMMA_N, WMMA_K, float> c_frag;
-  wmma::fill_fragment(c_frag, 0.0);
-  for (int step = 0; step < raft::ceildiv(data_dim, TILE_COL_WIDTH); step++) {
-    int num_load_elems = (step == raft::ceildiv(data_dim, TILE_COL_WIDTH) - 1)
-                           ? data_dim - step * TILE_COL_WIDTH
-                           : TILE_COL_WIDTH;
+  if (metric != cuvs::distance::DistanceType::BitwiseHamming) {
```

Comment:
The only diff here is wrapping the wmma operation part with `if (metric != cuvs::distance::DistanceType::BitwiseHamming)` so that we don't perform unnecessary computations for `BitwiseHamming`. (the diff seems to detect all additional indents as changes)